c10/util/qint8.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+#include <cstdint>
+
+namespace c10 {
+
+/**
+ * This is the data type for quantized Tensors. Right now we only have
+ * qint8 which is for 8 bit Tensors, we might have 4 bit, 2 bit or 1 bit
+ * data types in the future.
+ */
+struct alignas(1) qint8 {
+  uint8_t val_;
+  /* implicit */ qint8(uint8_t val=0) : val_(val) {}
+
+  // This conversion is here because we specified that qint8 has TAG_i
+  // See c10/core/Scalar.h for more details.
+  inline operator uint32_t() const {
+    return static_cast<uint32_t>(val_);
+  }
```

Comment:
Scalar.h is not modified in this diff, so I am not sure what extra explanation there is, but this seems very fishy to me. Just thinking of `qint8` as a data type (independent of Scalar), I don't want it to be implicitly convertible to `uint32_t`. That sounds like a bug farm. Can you explain more what the `Scalar` situation is?

Reply:
In DEFINE_ACCESSOR, we call checked_convert on type QInt8 and int64_t, QInt8 and double, this is to eliminate the compilation error from these calls.

Reply:
Maybe the right thing to do is to fix Converter here to support QInt types:https://github.com/pytorch/pytorch/blob/master/c10/util/Half.h#L410

Reply:
I'm planning to change this to conversion to uint8_t

Reply:
So what's the conclusion on this? can we get rid of implicit conversion here?